### PR TITLE
Download artifacts from partially succeeded builds

### PR DESCRIPTION
**Pull request by Thilas**
Per #8431, add option to DownloadBuildArtifactsV0 task in order to allow to download artifacts from partially succeeded builds.

### CODE REVIEW

**Comment by Thilas** [comment]
@omeshp, please review my PR.

FYI, I'm willing to add tests if required but I would just need some directions since there are none on the DownloadBuildArtifactsV0 task for now.

**Comment by Thilas** [comment]
@GitHubSriramB, @kasubram, if any of you could review this PR as well, I would be great. Thanks.

**Comment by Thilas** [comment]
@kmkumaran, any news about this?

**Comment by kmkumaran** [comment]
@Thilas  - @kasubram and @omeshp will own reviewing this PR. They will respond to you on this.

File: Tasks/DownloadBuildArtifactsV0/task.json [hunk 1]
```diff
@@ -84,6 +84,15 @@
                 "specific": "Specific version"
             }
         },
+        {
+            "name": "partiallySucceededBuilds",
```
Comment (Lovakumar) [inline Tasks/DownloadBuildArtifactsV0/task.json]: Should we rename this to allowPartiallySucceededBuilds and other places wherever we consume the same?

Comment (Thilas) [reply Tasks/DownloadBuildArtifactsV0/task.json]: Thanks for the review. I just renamed the parameter as requested.

**Comment by Thilas** [comment]
What's next? It has been approved so I guess it just need to be merged now, right? Thanks.
